src/Prefix.jl

```diff
@@ -388,22 +388,27 @@ function setup_dependencies(prefix::Prefix, dependencies::Vector{PkgSpec}, platf
     end
     dependencies = filter_redundant_version.(dependencies)
 
+    # Get julia version specificity, if it exists, from the `Platform` object
+    julia_version = nothing
+    if haskey(platform, "julia_version")
+        julia_version = p["julia_version"]
+    end
```

Comment:
```suggestion
    julia_version = get(platform, "julia_version", nothing)
```
?  Is the value of `julia_version` a string or a `VersionNumber`?

Reply:
Oh, that's a good point, it needs to be a `VersionNumber` for `Pkg`.